core/app/components/ActionUI/ActionRunForm.tsx

```diff
@@ -23,20 +23,29 @@ export const ActionRunForm = ({
 	actionInstance,
 	pubId,
 	fieldConfig,
+	defaultFields,
 }: {
 	actionInstance: ActionInstances;
 	pubId: PubsId;
 	fieldConfig: FieldConfig<any>;
+	defaultFields: string[];
 }) => {
-	const community = useCommunity();
-	const runAction = useServerAction(runActionInstance);
-	// const runAction = useServerAction(async (props: any) => {
-	// 	return { success: "true", report: "Wheee" };
-	// });
-
 	const [isPending, startTransition] = useTransition();
-
 	const action = getActionByName(actionInstance.action);
+	const schema = useMemo(() => {
+		const schemaWithPartialDefaults = (action.config.schema as z.ZodObject<any>).partial(
+			defaultFields.reduce(
+				(acc, key) => {
+					acc[key] = true;
+					return acc;
+				},
+				{} as Record<string, true>
+			)
+		);
+		return schemaWithPartialDefaults;
+	}, [action.config.schema, defaultFields]);
```

Comment:
This effectively makes the required fields optional when a default is supplied for a field.